### DISCUSSION TITLE
During the damage simulation, no damage result was output，My yaml file is as follows

### DISCUSSION THREAD

**Discussion by TAIYONGZOU** [damask-multiphysics/DAMASK#9] [Q&A] [answered]
homogenization:
  SX:
    N_constituents: 1
    mechanical: 
      type: pass
    damage:
      type: pass
      initialDamage: 1.0
      output: damage
      porosity:      phasefield
      output: porosity
     

phase:
  Aluminum:
    lattice: cF
    mechanical:
      output: [F, P, F_e, F_p, L_p, O]
      elastic: {type: Hooke, C_11: 107e9, C_12: 52e9, C_44: 26e9}
      plastic:
        type: dislotwin
        output: [rho_dip, rho_mob]
        D: 1.525e-05
        tau_0: [55e6]
        N_sl: [12]
        b_sl: [2.546e-10]
        rho_dip_0: [1]
        rho_mob_0: [1e12]
        v_0: [10e-5]
        Q_sl: [8.361e-20]
        Q_cl: 1.876e-19
        p_sl: [0.8]
        q_sl: [1.5]
        i_sl: [45]
        V_at: 6.6e-29
        D_a: 7
        D_0: 0.000623
        h_sl-sl: [1, 1, 1, 1, 1, 1, 1]
    damage:       
      - type: anisobrittle
        output: [f_phi]
        W_crit: 500000.0
        N_cl: [3]
        g_crit: [0.5e+7]
        s_crit: [0.006666]
        dot_o: 1.e-3
        l_c: 1.0
        q: 20
        mu: 0.001

material:
  - constituents:
      - O: [0.8824265782080677, -0.05608753307113829, 0.08883322634416188, -0.45856971182502]
        phase: Aluminum
        v: 1.0
    homogenization: SX
  - constituents:
      - O: [0.6134475557126102, 0.6543253967799157, 0.2798496271731801, -0.34238072038311673]
        phase: Aluminum
        v: 1.0
    homogenization: SX
  - constituents:
      - O: [0.7230577269230365, -0.5061658616635942, -0.4606239451877223, -0.09385747249388812]
        phase: Aluminum
        v: 1.0
    homogenization: SX
  - constituents:
      - O: [0.13098834197947687, -0.9552127374211582, 0.12172171101201634, 0.23578487144461188]
        phase: Aluminum
        v: 1.0
    homogenization: SX
  - constituents:
      - O: [0.7651127176405781, 0.3814195211793133, -0.4439972574587404, -0.26830600726988824]
        phase: Aluminum
        v: 1.0
    homogenization: SX

**Comment by hyharry** [comment]
Did you forget to set a solver for your damage field? Normally you can do that along with your mechanical solver like this in your `load.yaml` file. 

`solver: {mechanical: spectral_basic, damage: spectral}`

**Comment by TAIYONGZOU** [comment]
Thank you!

   In my results only the damage driving force (f_phi) is output, but there is no damage (phi) output.Is this because my material.yaml file settings are wrong？






---- Replied Message ----
| From | ***@***.***> |
| Date | 4/4/2024 22:26 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [eisenforschung/DAMASK] During the damage simulation, no damage result was output，My yaml file is as follows (Discussion #9) |

Did you forget to set a solver for your damage field? Normally you can do that along with your mechanical solver like this in your load.yaml file.

solver: {mechanical: spectral_basic, damage: spectral}

**Reply by hyharry** [reply]
@TAIYONGZOU Hello, if you think my answer solves your problem, please mark my reply as answer. This will also help the others.

**Comment by hyharry** [comment, marked as answer]
if you are using the official `damask3`, you can specify damage output in the `homogenization` block of your `material.yaml`, e.g.

```yaml
homogenization:
  SX:
    N_constituents: 1
    mechanical: {type: pass}
    damage:
      output: [phi]
      type: pass
```

**Comment by TAIYONGZOU** [comment]
I really appreciate your help！







---- Replied Message ----
| From | ***@***.***> |
| Date | 4/5/2024 21:13 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [eisenforschung/DAMASK] During the damage simulation, no damage result was output，My yaml file is as follows (Discussion #9) |

@TAIYONGZOU Hello, if you think my answer solves your problem, please mark my reply as answer. This will also help the others.

**Comment by TAIYONGZOU** [comment]
Dear DAMASK team：
     I want to simulate phase field damage, how should I set up the material.yaml file?







---- Replied Message ----
| From | ***@***.***> |
| Date | 4/4/2024 22:26 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [eisenforschung/DAMASK] During the damage simulation, no damage result was output，My yaml file is as follows (Discussion #9) |

Did you forget to set a solver for your damage field? Normally you can do that along with your mechanical solver like this in your load.yaml file.

solver: {mechanical: spectral_basic, damage: spectral}

**Reply by hyharry** [reply]
The damage simulation in DAMASK is already a phase-field formulation. Pls refer to the DAMASK paper.

**Comment by TAIYONGZOU** [comment]
I got it！ Thank you






---- Replied Message ----
| From | ***@***.***> |
| Date | 4/8/2024 17:31 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [eisenforschung/DAMASK] During the damage simulation, no damage result was output，My yaml file is as follows (Discussion #9) |

The damage simulation in DAMASK is already a phase-field formulation. Pls refer to the DAMASK paper.
